hathor/wallet/resources/thin_wallet/address_history.py

```diff
@@ -166,18 +166,20 @@ def get_address_history(self, addresses: list[str], ref_hash: Optional[str]) ->
                     'message': 'The address {} is invalid'.format(address)
                 })
 
-            tx = None
-            if ref_hash_bytes:
-                try:
-                    tx = self.manager.tx_storage.get_transaction(ref_hash_bytes)
-                except TransactionDoesNotExist:
-                    return json_dumpb({
-                        'success': False,
-                        'message': 'Hash {} is not a transaction hash.'.format(ref_hash)
-                    })
-
             # The address index returns an iterable that starts at `tx`.
-            hashes = addresses_index.get_sorted_from_address(address, tx)
+            if idx == 0:
+                ref_tx = None
+                if ref_hash_bytes:
```

Comment:
Maybe move this fetch to a separate method to clean up the code?

Reply:
I'd like to also refactor other parts of the code to make it easier to read and to handle errors and I think I'd rather do it all in a separate PR, what do you think?

Reply:
Sure!